curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/write-higher-order-arrow-functions.english.md

```diff
@@ -1,42 +1,73 @@
 ---
 id: 587d7b88367417b2b2512b45
-title: Write Higher Order Arrow Functions
+title: Write Higher-Order Arrow Functions
 challengeType: 1
 ---
 
 ## Description
 <section id='description'>
-It's time we see how powerful arrow functions are when processing data.
-Arrow functions work really well with higher order functions, such as <code>map()</code>, <code>filter()</code>, and <code>reduce()</code>, that take other functions as arguments for processing collections of data.
-Read the following code:
-<blockquote>FBPosts.filter(function(post) {<br>&nbsp;&nbsp;return post.thumbnail !== null && post.shares > 100 && post.likes > 500;<br>})</blockquote>
-We have written this with <code>filter()</code> to at least make it somewhat readable. Now compare it to the following code which uses arrow function syntax instead:
-<blockquote>FBPosts.filter((post) => post.thumbnail !== null && post.shares > 100 && post.likes > 500)</blockquote>
-This code is more succinct and accomplishes the same task with fewer lines of code.
+It's time to look at higher-order functions and their common pair, arrow functions.
+Arrow functions work really well when combined with higher-order functions, such as <code>map()</code>, <code>filter()</code>, and <code>reduce()</code>. <br>
+But what are these functions? Lets look at the simplest example <code>forEach()</code>, and run it on the following array of sample Facebook posts.
+<blockquote>let FBPosts = [<br>
+    {thumbnail: "someIcon", likes:432, shares: 600},<br>
+    {thumbnail: "Another icon", likes:300, shares: 501},<br>
+    {thumbnail: "Yet another", likes:40, shares: 550},<br>
+    {thumbnail: null, likes: 101, shares:0},<br>
+]</br>
+</blockquote>
+Of the two <code>forEach()</code> versions below, both perform the exact same log function, and each takes an anonymous callback with a parameter <code>post</code>. The difference is the syntax. One uses an arrow function and the other does not.  
+<blockquote>
+<strong>ES5</strong><br>
+FBpost.forEach(function(post) {<br>
+    console.log(post) // log each post here<br>
+    })<br>
+<strong>ES6</strong><br>
+FBpost.forEach((post) => {<br>
+    console.log(post) // log each post here<br>
+    })<br>
+
+</blockquote>
+<code>filter()</code> is very similar. Below it will iterate over the <code>FBPosts</code> array, perform the logic to filter out the items that do not meet the requirements, and return a new array <code>results</code>. All the higher-order JavaScript functions operate in a similar way.
```

Comment:
```Below it will iterate over the <code>FBPosts</code> array, perform the logic to filter out the items that do not meet the requirements, and return a new array, <code>results</code>```
Added comma before array at end.